azurerm/resource_arm_policy_assignment.go

```diff
@@ -57,6 +58,34 @@ func resourceArmPolicyAssignment() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"location": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				ForceNew:         true,
+				StateFunc:        azure.NormalizeLocation,
+				DiffSuppressFunc: azure.SuppressLocationDiff,
+			},
```

Comment:
rather than defining this here could we move this logic into a new function in `./azurerm/helpers/azure/location.go` called `SchemaLocationOptional`? we should then be able to reference this as `"location": locationSchemaOptional()`